### PR TITLE
[W06H01] improve loop finding

### PR DESCRIPTION
**Pull request by felixhauptmann**
* walk all .java files in src directory
* output line number of offending keyword

### CODE REVIEW

File: w06h01/test/pgdp/LoopTest.java [hunk 1]
```diff
@@ -62,7 +79,7 @@ private String removeMultipleMethods(String file, String[] methods) {
 
     /**
      * Removes a method from a file
-     *
+     * <p>
```
Comment (nilsreichardt) [inline w06h01/test/pgdp/LoopTest.java:82]: For what is this `<p>`?

File: w06h01/test/pgdp/LoopTest.java [hunk 2]
```diff
@@ -18,39 +21,53 @@ public class LoopTest {
...
+        Pattern pattern = Pattern.compile(regExp);
+
+        String[] lines = s.split("\n");
+        for (int i = 0; i < lines.length; i++) {
+            Matcher matcher = pattern.matcher(lines[i]);
+            if (matcher.find()) {
+                return i + 1;
             }
         }
+        return -1;
```
Comment (nilsreichardt) [inline w06h01/test/pgdp/LoopTest.java:70]: Just a small comment to indicate what `-1` means

```suggestion
        // No loop found, returning -1
        return -1;
```

**Review by nilsreichardt — CHANGES_REQUESTED** [review]
Besides the comments, LGTM ❤️ Nice improvements!

File: w06h01/test/pgdp/LoopTest.java (hunk 1 above)
Comment (julian-hartl) [reply w06h01/test/pgdp/LoopTest.java:82]: You can embed html tags in java docs. Just makes it look prettier.

**Review by MaximilianAnzinger — APPROVED** [review]
[🐧](https://emojipedia.org/penguin/)

**Review by MaximilianAnzinger — APPROVED** [review]
[🐧](https://emojipedia.org/penguin/)

**Pull request merged**